lib/hooks/use-product-results.ts

```diff
@@ -0,0 +1,62 @@
+import { SerializeFrom } from '@remix-run/node';
+import { useEffect, useState } from 'react';
+import { getEcomApi, IProductFilters, Product, ProductSortBy, Slice } from '../ecom';
+import { getErrorMessage } from '../utils';
+
+export interface UseProductResultsArgs {
+    categorySlug: string;
+    filters: IProductFilters;
+    sorting: ProductSortBy;
+    initialResultsFromLoader: Slice<Product | SerializeFrom<Product>>;
+}
+
+/**
+ * Manages product slicing and loads next slices from the client side
+ * without a full page reload. Syncs with initialResultsFromLoader.
```

Comment:
useProductsPageResults()

Returns the list of products displayed on the products page.


initialResultsFromLoader - the first batch of results loaded by the route loader to avoid fetching it again client-side.